### PR TITLE
Issue #929: fix set charset for request content

### PR DESCRIPTION
**Pull request by phos-web**
Get encoding charset from HTTP headers or fallback to UTF-8

### CODE REVIEW

File: endpoints/citrus-http/src/test/java/org/citrusframework/http/interceptor/LoggingClientInterceptorTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    @Test
+    public void testGetRequestContentWithNullContentType() throws Exception {
+        HttpRequest request = Mockito.mock(HttpRequest.class);
+        HttpHeaders headers = Mockito.mock(HttpHeaders.class);
+
+        when(request.getMethod()).thenReturn(HttpMethod.valueOf("GET"));
+        when(request.getURI()).thenReturn(new URI("http://example.com"));
+        when(request.getHeaders()).thenReturn(headers);
+        when(headers.getFirst(HttpHeaders.CONTENT_TYPE)).thenReturn(null);
```
Comment (bbortt) [inline endpoints/citrus-http/src/test/java/org/citrusframework/http/interceptor/LoggingClientInterceptorTest.java:76]: try this as another test case, please: `when(headers.getFirst(HttpHeaders.CONTENT_TYPE)).thenReturn("");`

File: endpoints/citrus-http/src/main/java/org/citrusframework/http/interceptor/LoggingClientInterceptor.java [hunk 2]
```diff
@@ -120,6 +121,24 @@ public boolean hasMessageListeners() {
...
+        String contentType = request.getHeaders().getFirst(HttpHeaders.CONTENT_TYPE);
+        if (contentType != null) {
+            String[] contentTypeParts = contentType.split(";");
+            for (String contentTypePart : contentTypeParts) {
+                if(contentTypePart.startsWith("charset=")) {
+                    String charset = contentTypePart.split("=")[1];
+                    try {
+                        body = new String(body.getBytes(), charset);
+                    } catch (UnsupportedEncodingException e) {
+                        body = new String(body.getBytes(), StandardCharsets.UTF_8);
```
Comment (bbortt) [inline endpoints/citrus-http/src/main/java/org/citrusframework/http/interceptor/LoggingClientInterceptor.java:133]: please use static imports wherever possible.

```suggestion
                        body = new String(body.getBytes(), UTF_8);
```

File: endpoints/citrus-http/src/main/java/org/citrusframework/http/interceptor/LoggingClientInterceptor.java [hunk 3]
```diff
@@ -120,6 +121,24 @@ public boolean hasMessageListeners() {
      * @return
      */
     private String getRequestContent(HttpRequest request, String body) {
+        String contentType = request.getHeaders().getFirst(HttpHeaders.CONTENT_TYPE);
+        if (contentType != null) {
+            String[] contentTypeParts = contentType.split(";");
+            for (String contentTypePart : contentTypeParts) {
+                if(contentTypePart.startsWith("charset=")) {
+                    String charset = contentTypePart.split("=")[1];
```
Comment (bbortt) [inline endpoints/citrus-http/src/main/java/org/citrusframework/http/interceptor/LoggingClientInterceptor.java:129]: if for some reason the header contains only `charset=`? this line will fail. please add a check `contentTypePart.split("=").length == 2`.

**Review by bbortt — CHANGES_REQUESTED** [review]
thank you. I think one or two more checks would be valid. but it shouldn't be far from the final solution.

closes https://github.com/citrusframework/citrus/issues/929.

File: endpoints/citrus-http/src/test/java/org/citrusframework/http/interceptor/LoggingClientInterceptorTest.java (hunk 1 above)
Comment (phos-web) [reply endpoints/citrus-http/src/test/java/org/citrusframework/http/interceptor/LoggingClientInterceptorTest.java:76]: added this as another test case called `testGetRequestContentWithEmptyStringContentType()`

File: endpoints/citrus-http/src/main/java/org/citrusframework/http/interceptor/LoggingClientInterceptor.java (hunk 2 above)
Comment (phos-web) [reply endpoints/citrus-http/src/main/java/org/citrusframework/http/interceptor/LoggingClientInterceptor.java]: added static imports

File: endpoints/citrus-http/src/main/java/org/citrusframework/http/interceptor/LoggingClientInterceptor.java (hunk 3 above)
Comment (phos-web) [reply endpoints/citrus-http/src/main/java/org/citrusframework/http/interceptor/LoggingClientInterceptor.java:129]: prevent failure by adding `!contentTypePart.endsWith("charset=")` to if clause.

**Review by bbortt — APPROVED** [review]

**Pull request merged**